plugin-build/src/main/kotlin/io/sentry/android/gradle/telemetry/SentryTelemetryService.kt

```diff
@@ -0,0 +1,474 @@
+import javax.inject.Inject
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.internal.tasks.execution.ExecuteTaskBuildOperationDetails
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
+import org.gradle.api.provider.ValueSource
+import org.gradle.api.provider.ValueSourceParameters
+import org.gradle.api.services.BuildService
+import org.gradle.api.services.BuildServiceParameters.None
+import org.gradle.api.tasks.Input
+import org.gradle.execution.RunRootBuildWorkBuildOperationType
+import org.gradle.internal.operations.BuildOperationDescriptor
+import org.gradle.internal.operations.BuildOperationListener
+import org.gradle.internal.operations.OperationFinishEvent
+import org.gradle.internal.operations.OperationIdentifier
+import org.gradle.internal.operations.OperationProgressEvent
+import org.gradle.internal.operations.OperationStartEvent
+import org.gradle.process.ExecOperations
+import org.gradle.util.GradleVersion
+
+abstract class SentryTelemetryService :
+    BuildService<None>,
+    BuildOperationListener,
+    AutoCloseable {
+
+    private var hub: IHub = NoOpHub.getInstance()
+    private var transaction: ITransaction? = null
+    private var didAddChildSpans: Boolean = false
+    private var started: Boolean = false
+
+    fun start(startParameters: SentryTelemetryServiceParams) {
+        if (started) {
+            return
+        }
+
+        started = true
+        try {
+            if (startParameters.saas == false) {
+                SentryPlugin.logger.info(
+                    "Sentry is running against a self hosted instance. Telemetry has been disabled."
+                )
+                hub = NoOpHub.getInstance()
+            } else if (!startParameters.sendTelemetry) {
+                SentryPlugin.logger.info("Sentry telemetry has been disabled.")
+                hub = NoOpHub.getInstance()
+            } else {
+                if (!Sentry.isEnabled()) {
+                    SentryPlugin.logger.info(
+                        "Sentry telemetry is enabled. To disable set `telemetry=false` " +
+                            "in the sentry config block."
+                    )
+                    Sentry.init { options ->
+                        options.dsn = startParameters.dsn
+                        options.isDebug = true
+                        options.isEnablePrettySerializationOutput = false
+                        options.tracesSampleRate = 1.0
+                        options.release = BuildConfig.Version
+                        options.isSendModules = false
+                        options.environment = startParameters.buildType
+                        options.setTag("SDK_VERSION", BuildConfig.SdkVersion)
+                        options.setTag("AGP_VERSION", AgpVersions.CURRENT.toString())
+                        options.setTag("BUILD_SYSTEM", "gradle")
+                        options.setTag("BUILD_TYPE", startParameters.buildType)
```

Comment:
`m`: do we need it as a tag, if we already send it as `environment` which is also indexed and searchable?

Reply:
wasn't sure how this'll be queryable in dashboards

Reply:
yeah environment is queryable, so we probably don't need it to save our own space